src/sentry/eventstream/kafka/protocol.py

```diff
@@ -211,6 +211,13 @@ def get_task_kwargs_for_message_from_headers(
                 )
             task_state["group_states"] = group_states
 
+            # default in case queue is not sent
+            task_state["queue"] = (
+                decode_str(header_data["queue"])
```

Comment:
do we need to add queue info in case it doesn't get the info from headers from `get_task_kwargs_for_message`?

Reply:
Yeah you will need to include the queue in the message body as well so that it can be fetched from `get_task_kwargs_for_message`. Somewhere around here: https://github.com/getsentry/sentry/blob/7a28bc4ef2fa8088c860844359869460552379e0/src/sentry/eventstream/snuba.py#L182-L188